ui/components/report.jsx

```diff
@@ -1,36 +1,58 @@
-var React = require('react'),
-    Header = require('./header.jsx'),
-    Results = require('./results.jsx');
+import React from 'react';
+import Header from './header.jsx';
+import Results from './results.jsx';
 
 function _hasPassed(test) {
   return test.state === 'passed';
 }
 
 var Report = React.createClass({
+  getInitialState: function() {
+    return {
+      filter: 'all'
+    };
+  },
+  updateFilter: function(newFilter) {
+    this.setState({
+      filter: newFilter
+    });
+  },
   render: function() {
     var suites = this.props.data,
         passes = [],
         failures = [],
+        numberOfPasses = 0,
+        numberOfFailures = 0,
         duration = 0;
 
     suites.forEach(function(suite) {
-      passes = passes.concat(suite.tests.filter(function(test) {
+      passes = [];
+      failures = [];
+
+      suite.passes = passes.concat(suite.tests.filter(function(test) {
```

Comment:
You can do `suite.passes = suite.tests.filter(function(test){ ...` directly and drop all the passes/failures arrays